### ISSUE TITLE
Timing to hit the new 5 min api - how many seconds into each interval ?

### ISSUE THREAD

**Issue by craigcurtin-dev** [amberelectric/public-api#229]
Myles

Without overloading the system what is a (reasonably) safe time to hit the 5min api to get the numbers as soon as possible into the interval (i have my own automations in Node Red and hence touch my inverters directly) 

Is there a failure result (to enable a tight retry) if the number for that current interval has not been published or will it just return the previous interval ?

regards

Craig

**Comment by madpilot** [comment]
Once a minute, though if you offset by 30s, you should get new prices pretty quickly.

**Comment by craigcurtin-dev** [comment]
But based on 5 minute settlements that means I have lost 20% of the time window ?

Maybe I did not word it properly – say 10:00 – do I just wait 30s to make the call or is it variable ?

And does it give a failure message or will it just return the previous one if the new one is not yet published ?

Craig

**Comment by madpilot** [comment]
The prices don't usually hit our database until around 15-30s in the interval. You could try doing a request every 5 seconds between 15s and 35s and you should be good.

I'm costing up something a bit more push at the moment.

**Comment by craigcurtin-dev** [comment]
OK will give it a go and see how we go

Thanks for all you do on this stuff

Craig

**Comment by craigcurtin-dev** [comment]
15s seems to be hitting well at the moment and returning the expected info - i am trapping anything that is not a HTTP 200 and also recording each of the value and times returned - will try and do some analysis over the weekend to see if there have been any non-visable failures (i.e. return the previous block because not updated 

Craig

**Comment by purcell-lab** [comment]
Amber2MQTT does a great job of grabbing the prices very quickly and then backing off until the next 5 minute interval.

https://github.com/cabberley/amber2mqtt

**Comment by kurtgetonwithit** [comment]
> 15s seems to be hitting well at the moment and returning the expected info - i am trapping anything that is not a HTTP 200 and also recording each of the value and times returned - will try and do some analysis over the weekend to see if there have been any non-visable failures (i.e. return the previous block because not updated
> 
> Craig

I've only recently started working on this issue for myself. What I've found is working so far is starting to check immediately after the previous price data's nemTime has passed and looping every 5 seconds. If the prices come back with the 'estimate' attribute being true, then loop again. I might change it to 15 seconds past the nemTime, but am currently no where near hitting the throttling limits of 50 per 300 seconds.

**Comment by craigcurtin-dev** [comment]
Yes where you will hit the API limit is if you get repeated failures and keep trying.

It has been very stable lately so no dramas.

Someone has also developed a Docker image that does all of this with auto retries etc and also falls back to the NEM/AEMO feed

I intend to try and make some time to try and check it out and see whether it is worth switching over and just ingest the prices through MQTT

Craig

**Comment by jeffborg** [comment]
> Someone has also developed a Docker image that does all of this with auto retries etc and also falls back to the NEM/AEMO feed

It amber2mqtt does not actually fall back, it simply does the same checking for aemo data, you would need to do some smarts to get the price from the fastest response in the 5 minute window, amber usually beats aemo.
